frontend/packages/console-app/src/components/tour/TourStepComponent.tsx

```diff
@@ -59,20 +59,20 @@ const TourStepComponent: React.FC<TourStepComponentProps> = ({
   const footer = (
     <StepFooter
       primaryButton={{
-        name: nextButtonText,
+        name: nextButtonText || '',
         onClick: () => {
           onNext && onNext();
         },
       }}
       secondaryButton={{
-        name: backButtonText,
+        name: backButtonText || '',
         onClick: () => {
           onBack && onBack();
         },
       }}
       step={step}
     >
-      {showStepBadge ? <StepBadge stepNumber={step} totalSteps={totalSteps} /> : null}
+      {showStepBadge ? <StepBadge stepNumber={step ?? 0} totalSteps={totalSteps ?? 0} /> : null}
```

Comment:
nit: The props default values might be cleaner than Nullish Coalescing Operator.